semgrep-core/src/parsing/other/yaml_to_generic.ml

```diff
@@ -168,65 +171,80 @@ let do_parse env =
 (* Parser functions *)
 (*****************************************************************************)
 
-let make_alias anchor pos env : G.name = mk_id anchor pos env
+let make_node f anchor args env =
+  let node_expr = f args env in
+  (match anchor with
+  | Some anchor -> Hashtbl.add env.anchors anchor node_expr
+  | None -> ());
+  node_expr
+
+let make_alias anchor pos env =
+  let t = mk_tok pos anchor env in
+  match Hashtbl.find_opt env.anchors anchor with
+  | Some (expr, _p) -> (G.e (G.Alias ((anchor, t), expr)), pos)
+  | None -> raise UnrecognizedAlias
```

Comment:
I would use an `option` or `result` type here to be on the safe side. E.g. using `result` and polymorphic variants you could write this as:
```ocaml
let make_alias anchor pos env =
   let t = mk_tok pos anchor env in
   match Hashtbl.find_opt env.anchors anchor with
   | Some (expr, _p) -> Ok (G.e (G.Alias ((anchor, t), expr)), pos)
   | None -> Error `UnrecognizedAlias
```